jbmc/src/java_bytecode/java_bytecode_convert_class.cpp

```diff
@@ -1197,19 +1198,21 @@ void mark_java_implicitly_generic_class_type(
   // implicitly generic and update identifiers of type parameters used in fields
   if(!implicit_generic_type_parameters.empty())
   {
-    class_symbol.type = java_implicitly_generic_class_typet(
+    java_implicitly_generic_class_typet new_class_type(
       class_type, implicit_generic_type_parameters);
 
-    for(auto &field : class_type.components())
+    for(auto &field : new_class_type.components())
     {
       find_and_replace_parameters(
         field.type(), implicit_generic_type_parameters);
     }
 
-    for(auto &base : class_type.bases())
+    for(auto &base : new_class_type.bases())
     {
       find_and_replace_parameters(
         base.type(), implicit_generic_type_parameters);
     }
+
+    class_symbol.type = new_class_type;
```

Comment:
! This which was in the Code Cleanup PR suddenly looks like a bugfix -- the results of `find_and_replace_parameters` would previously have been wrongly discarded, right? Can we test the fix?

Reply:
My assumption was that the thing we were changing was a reference to the same and this was just clarifying things. I'll check.

Reply:
It was. I've updated the commit message to make this clear.